### ISSUE TITLE
Bluetooth beacon blocks

### ISSUE THREAD

**Issue by bluetooth-mdw** [Microsoft/pxt-microbit#115]
It would be really useful if a developer could turn their micro:bit into a Bluetooth beacon merely by dragging and dropping a sing block. It's requested that new blocks be created to make this possible.

Note that there are three types of beacon and associated data format prevalent in the world today:

iBeacon from Apple

AltBeacon - an open source format similar to iBeacon

EddyStone from google

All three use Bluetooth advertising packets to broadcast small amounts of data, often, but not always for the purpose of encoding an indoor location.

iBeacon and AltBeacon both use a unique identifier which is intended to map to a location of some kind.

EddyStone has three sub-frame types, one which expresses a UID like iBeacon, one which encodes a URI and one which encodes beacon telemetry data.

I believe Lancaster University has a fork of the runtime in which beacon APIs are being created.

It may be too early to progress this item immediately, therefore but I wanted to get this issue logged now whilst I think of it.

**Comment by tballmsft** [comment]
Thanks, Martin. I'm going to create a Bluetooth Milestone for these.

**Comment by bluetooth-mdw** [comment]
Great, thank you :-)

**Comment by tballmsft** [comment]
Martin, are you able to code/test some of these features yourself? It would be great to have your help.

**Comment by bluetooth-mdw** [comment]
I sure can and would be glad to help.

**Comment by tballmsft** [comment]
Martin, you should dig into how we expose C++ runtime functions as blocks in PXT. I suspect you will do a lot of you work there.   Likely, you will not need to write much, if any, TypeScript. See https://www.pxt.io/simshim

**Comment by bluetooth-mdw** [comment]
Agreed. You presumably know James D at Lancaster Uni? He's been working on adding beacon support (EddyStone specifically) in the runtime in a branch. He's away on vacation so I'll pick up with him when he gets back. i had a quick look at the code yesterday and it's not quite there yet. So I think this issue can be considered a placeholder until we have things in place in the runtime.

It will be awesome when we have it though! Drag. Drop. Tweak. Bang! My micro:bit's a beacon! :-)
